src/lib/parser.rs

```diff
@@ -108,8 +108,51 @@ impl Parser {
     Ok(parser)
   }
   pub fn next_char(&mut self) -> Option<char> {
+    // get next
```

Comment:
Some of the comments are not really necessary, i like comments very much but sometimes things just explain itself :)  
Some comments that are not really needed as the code already explains them:   
- define forward slash, newline & astrix
- increase index *(It's better to add a comment to the index field in the struct so you can just click to go to it's definition and see what it's all about)*

Some comments i very much like:  
- check for end line *(Maybe you can change this comment to also include the next comment after it)*

Reply:
I've made some changes. Let me know if you think there is anything else